syft/frameworks/torch/mpc/falcon/falcon.py

```diff
@@ -11,3 +14,71 @@ def conv2d(filters: syft.ReplicatedSharingTensor, image, padding=0):
     output_size = (image_height - filter_height + 2 * padding) + 1
     result = result.view(-1, channels_out, output_size, output_size)
     return result
+
+
+def select_share(
+    b_share: ReplicatedSharingTensor,
+    x_share: ReplicatedSharingTensor,
+    y_share: ReplicatedSharingTensor,
+) -> ReplicatedSharingTensor:
+    """Performs select share protocol
+    If the bit b_share is 0, x_sh is returned
+    If the bit b_share is 1, y_sh is returned
+
+    Args:
+        x_share (ReplicatedSharingTensor): the first share to select
+        y_share (ReplicatedSharingTensor): the second share to select
+        b_share (ReplicatedSharingTensor): the bit to take into consideration
+            when selecting between x or y
+
+    Return:
+        x_share or y_share (both are ReplicatedSharingTensor): the b_share
+        tells us what share to select
+    """
+
+    ring_size = x_share.ring_size
+    players = x_share.players
+
+    # This should be generated by a crypto provider or create a 0 RST shared value and make
+    # each worker add a random number at an offline step
+    c = torch.randint(high=2, size=(1,))
+    c_share_2 = c.share(*players, protocol="falcon", field=2)
+    c_share_L = c.share(*players, protocol="falcon", field=ring_size)
+
+    xor_b_c = FalconHelper.xor(b_share, c_share_2)
+
+    # TODO: Think of a method to do this better (1 - c_share) without being to
+    # "intrusive" with the operation
+    ones_share = ReplicatedSharingTensor.one_shares(players, ring_size)
+    if xor_b_c.reconstruct().item():
+        d_share = ones_share - c_share_L
+    else:
+        d_share = c_share_L
+
+    selected_share = (y_share - x_share) * d_share + x_share
+    return selected_share
+
+
+def evaluate(
```

Comment:
if you noticed this routine determines the negativity of x, if beta is an even number this will return x, if beta is odd it will return - x, you can choose a name that explains this functionality

Reply:
Will rename it